ayon_api/entity_hub.py

```diff
@@ -1485,6 +1486,704 @@ def fill_children_ids(self, children_ids):
+        self._original_index = index
+
+        self._index = index
+        self._project_statuses = project_statuses
+        self._is_new = index is None or project_statuses is None
+
+    def __str__(self):
+        short_name = ""
+        if self.short_name:
+            short_name = "({})".format(self.short_name)
+        return "<{} {}{}>".format(
+            self.__class__.__name__, self.name, short_name
+        )
+
+    def __repr__(self):
+        return str(self)
+
+    def __getitem__(self, key):
+        if key in {
+            "name", "short_name", "icon", "state", "color", "slugified_name"
+        }:
+            return getattr(self, key)
+        raise KeyError(key)
+
+    def __setitem__(self, key, value):
+        if key in {"name", "short_name", "icon", "state", "color"}:
+            return setattr(self, key, value)
+        raise KeyError(key)
+
+    def lock(self):
+        """Lock statuse.
+
+        Changes were commited and current values are now the original values.
+        """
+
+        self._is_new = False
+        self._original_name = self.name
+        self._original_short_name = self.short_name
+        self._original_icon = self.icon
+        self._original_state = self.state
+        self._original_color = self.color
+        self._original_index = self.index
+
+    @staticmethod
+    def slugify_name(name):
+        """Slugify status name for name comparison.
+
+        Args:
+            name (str): Name of the status.
+
+        Returns:
+            str: Slugified name.
+        """
+
+        return slugify_string(name.lower())
+
+    def get_project_statuses(self):
+        """Internal logic method.
+
+        Returns:
+            _ProjectStatuses: Project statuses object.
+        """
+
+        return self._project_statuses
+
+    def set_project_statuses(self, project_statuses):
+        """Internal logic method to change parent object.
+
+        Args:
+            project_statuses (_ProjectStatuses): Project statuses object.
+        """
+
+        self._project_statuses = project_statuses
+
+    def unset_project_statuses(self, project_statuses):
+        """Internal logic method to unset parent object.
+
+        Args:
+            project_statuses (_ProjectStatuses): Project statuses object.
+        """
+
+        if self._project_statuses is project_statuses:
+            self._project_statuses = None
+            self._index = None
+
+    @property
+    def changed(self):
+        """Status has changed.
+
+        Returns:
+            bool: Status has changed.
+        """
+
+        return (
+            self._is_new
+            or self._original_name != self._name
+            or self._original_short_name != self._short_name
+            or self._original_index != self._index
+            or self._original_state != self._state
+            or self._original_icon != self._icon
+            or self._original_color != self._color
+        )
+
+    def delete(self):
```

Comment:
Might be better to name it: `delete_project_statuses`  or `reset`? 

Feels weird calling it without argument:
```py
new_status = project.statuses
new_status.delete()
```

But it's deleting the class itself?